app/services/league/controllers/leagueController.js

```diff
@@ -0,0 +1,147 @@
+        if (!result.settings.public) {
+            if (!(leagueKey === req.body.leagueKey)) return res.status(400).send('Error: You tried to join a private league with an invalid key');
+        }
+        next();
+    });
+};
+
+exports.createLeague = async (req, res) => {
+    const user = res.locals.username;
+    const league = new League({
+        leagueName: req.body.leagueName,
+        leagueManager: user,
+        leagueKey: req.body.leagueKey,
+        settings: req.body.settings,
+    });
+
+    try {
+        const savedLeague = await league.save();
+        addPlayerToLeague(user, savedLeague._id).then(() => {
+            res.send(savedLeague);
+        });
+    } catch (err) {
+        res.status(400).send(err);
+    }
+};
+
+exports.joinLeague = async (req, res) => {
+    const { _id } = res.locals.league;
+    const { username } = res.locals;
+
+    addPlayerToLeague(username, _id)
+        .then((succ) => {
+            if (!res.locals.league.playerList.includes(username)) {
+                return res.send(`Sucessfully joined league! (${req.body.leagueName}) \n ${succ}`);
+            }
+            throw new Error(`Cannot join league that you are already a part of! (${req.body.leagueName}) \n`);
+        })
+        .catch((err) => res.status(422).send(`{${err}}`));
+};
+
+exports.leaveLeague = async (req, res) => {
+    const { _id } = res.locals.league;
+    const { username } = res.locals;
+
+    const request = Promise.all([
+        await League.findByIdAndUpdate(
+            _id,
+            { $pull: { playerList: username } },
+            { new: true },
+            (err) => {
+                if (err) throw err;
+            },
+        ),
+        await User.findOneAndUpdate(
+            { username },
+            { $pull: { leagues: _id } },
+            { new: true },
+            (err) => {
+                if (err) throw err;
+            },
+        ),
+    ]);
+    request
+        .then((succ) => {
+            if (res.locals.league.playerList.includes(username)) {
+                return res.send(`Successfully left league (${req.body.leagueName})\n ${succ}`);
+            }
+            throw new Error(`Cannot leave league! Please make sure that you are a part of the league, (${req.body.leagueName}) before attempting to leave`);
+        })
+        .catch((err) => res.status(422).send(`{${err}}`));
+};
+
+exports.getLeagues = async (req, res) => {
+    await League.find({}, (err, result) => {
+        if (err) throw err;
+        else res.send(result);
+    });
+};
+
+exports.getLeagueNames = async (req, res) => {
+    await League.find({}, 'leagueName', (err, result) => {
+        if (err) throw err;
+        res.send(result);
+    });
+};
+
+exports.disbandLeague = async (req, res) => {
+    const { _id, leagueManager: manager } = res.locals.league;
+    const { username } = res.locals;
+
+    if (manager !== username) {
```

Comment:
Not sure if we need to handle upper/lowercase - I think we should be good

Reply:
I think we're fine here as well